CorsixTH/Lua/humanoid_actions/seek_reception.lua

```diff
@@ -23,20 +23,44 @@ class "SeekReceptionAction" (HumanoidAction)
 ---@type SeekReceptionAction
 local SeekReceptionAction = _G["SeekReceptionAction"]
 
--- distance from queue that patient can join a reception desk queue
+-- Distance from queue that patient can join a reception desk queue.
 local can_join_queue_distance = 6
 
 function SeekReceptionAction:SeekReceptionAction()
   self:HumanoidAction("seek_reception")
 end
 
+--! Can the humanoid join the reception queue at the given position?
+--!param humanoid (Humanoid) Humanoid going to the reception.
+--!param x (int) X coordinate of the tile to consider.
+--!param y (int) Y coordinate of the tile to consider.
+--!param dist (int) Distance of the tile to the reception.
+--!return (bool) Whether the humanoid can start queueing.
 local function can_join_queue_at(humanoid, x, y, dist)
   local flag_cache = humanoid.world.map.th:getCellFlags(x, y)
   return flag_cache.hospital and not flag_cache.room and
       dist <= can_join_queue_distance and
       flag_cache.owner == humanoid.hospital:getPlayerIndex()
 end
 
+--! Trim the last part of the path when it is possible to join the reception queue.
+--!param humanoid Humanoid that will walk the path.
+--!param path_x (array of int) X coordinates of the path, modified in-place.
+--!param path_y (array of int) Y coordinates of the path, modified in-place.
+local function trimQueuingTail(humanoid, path_x, path_y)
+  local pathindex = #path_x
+  for i = pathindex - 1, 2, -1 do
+    if can_join_queue_at(humanoid, path_x[i], path_y[i], pathindex - i) then
+      path_x[i + 1] = nil
+      path_y[i + 1] = nil
+    else
+      break
+    end
+  end
+  return
```

Comment:
Is the return necessary here?

Reply:
Doesn't look like it

Reply:
Plan to change it? Will merge as is otherwise.